src/zino/tasks/bgpstatemonitortask.py

```diff
@@ -0,0 +1,420 @@
+            bgp_info = await self._get_general_bgp_info()
+
+        if not bgp_info:
+            return
+
+        bgp_info = self._fixup_ip_addresses(bgp_info=bgp_info)
+
+        for oid, result in bgp_info.items():
+            self._update_single_bgp_entry(oid=oid, row=result, local_as=local_as, uptime=uptime)
+
+    async def _get_bgp_style(self) -> Union[BgpStyle, None]:
+        snmp = SNMP(self.device)
+
+        # TODO Should this be moved to the SNMP file?
+        def _is_in_scope(entry: MibObject, oid: Tuple[str, str]):
+            object_type = snmp._oid_to_object_type(*oid)
+            snmp._resolve_object(object_type=object_type)
+            root = OID(object_type[0])
+            return root.is_a_prefix_of(other=entry.oid)
+
+        try:
+            entry = await snmp.getnext("BGP4-V2-MIB-JUNIPER", "jnxBgpM2")
+        except NoSuchNameError:
+            pass
+        else:
+            if entry and _is_in_scope(entry=entry, oid=("BGP4-V2-MIB-JUNIPER", "jnxBgpM2")):
+                return "juniper"
+
+        try:
+            entry = await snmp.getnext("CISCO-BGP4-MIB", "cbgpPeer2Table")
+        except NoSuchNameError:
+            pass
+        else:
+            if entry and _is_in_scope(entry=entry, oid=("CISCO-BGP4-MIB", "cbgpPeer2Table")):
+                return "cisco"
+
+        try:
+            entry = await snmp.getnext("BGP4-MIB", "bgp")
+        except NoSuchNameError:
+            pass
+        else:
+            if entry and _is_in_scope(entry=entry, oid=("BGP4-MIB", "bgp")):
+                return "general"
+
+        return None
+
+    async def _get_local_as(self, bgp_style: BgpStyle) -> Union[int, None]:
+        snmp = SNMP(self.device)
+        if bgp_style == "juniper":
+            # Juniper has multiple entries for this, so we just take the first
+            try:
+                return (await snmp.getnext("BGP4-V2-MIB-JUNIPER", "jnxBgpM2PeerLocalAs")).value
+            except NoSuchNameError:
+                _logger.info(f"router {self.device.name} misses jnxBgpM2PeerLocalAs")
+                return None
+        if bgp_style == "cisco":
+            try:
+                return (await snmp.get("CISCO-BGP4-MIB", "cbgpLocalAs", 0)).value
+            except NoSuchNameError:
+                _logger.info(f"router {self.device.name} misses cbgpLocalAs")
+                return None
+        if bgp_style == "general":
+            try:
+                return (await snmp.get("BGP4-MIB", "bgpLocalAs", 0)).value
+            except NoSuchNameError:
+                _logger.info(f"router {self.device.name} misses bgpLocalAs")
+                return None
+
+    async def _get_uptime(self) -> Union[int, None]:
+        snmp = SNMP(self.device)
+        try:
+            return (await snmp.get("SNMPv2-MIB", "sysUpTime", 0)).value / 100
+        except NoSuchNameError:
+            _logger.info(f"router {self.device.name} misses sysUpTime")
+            return None
+
```

Comment:
The `LinkStateTask` does more or less the same thing.. here's an opportunity to make some common function out of it...

Reply:
Where would I best move that function to?

Reply:
Good question. I might say that you don't need to fetch sysUptime, as it has already been fetched by the linkstatetask immediately prior to this, and added to the device state as the boot time.

But, then again, in some cases you might need an accurate and up-to-date value.  It could be argued that `_get_uptime()` would be such a common operation that maybe it should even be in the `Task` base class itself?